src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -162,72 +121,98 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 return;
             }
 
+            var declaration = (StatementSyntax)declarator.Parent.Parent;
+
             // If we convert this to 'if (o is Type x)' then 'x' will not be definitely assigned 
-            // in the Else branch of the IfStatement, or after the IfStatement.  Make sure 
+            // in the Else branch of the IfStatement, or after the IfStatement. Make sure 
             // that doesn't cause definite assignment issues.
-            if (IsAccessedBeforeAssignment(syntaxContext, declarator, ifStatement, cancellationToken))
+            if (!CheckDefiniteAssignment(semanticModel, localSymbol, declaration, statement, cancellationToken))
             {
                 return;
             }
 
             // Looks good!
             var additionalLocations = ImmutableArray.Create(
-                localDeclarationStatement.GetLocation(),
-                ifStatement.GetLocation(),
-                condition.GetLocation(),
-                initializerValue.GetLocation());
+                declaration.GetLocation(),
+                statement.GetLocation(),
+                leftmost.GetLocation(),
+                asExpression.GetLocation());
 
             // Put a diagnostic with the appropriate severity on the declaration-statement itself.
             syntaxContext.ReportDiagnostic(Diagnostic.Create(
-                GetDescriptorWithSeverity(severity),
-                localDeclarationStatement.GetLocation(),
+                GetDescriptorWithSeverity(styleOption.Notification.Value),
+                declaration.GetLocation(),
                 additionalLocations));
         }
 
-        private bool IsAccessedBeforeAssignment(
-            SyntaxNodeAnalysisContext syntaxContext,
-            VariableDeclaratorSyntax declarator,
-            IfStatementSyntax ifStatement,
+        private static bool CheckDefiniteAssignment(
+            SemanticModel semanticModel,
+            ISymbol localVariable,
+            StatementSyntax declaration,
+            StatementSyntax statement,
             CancellationToken cancellationToken)
         {
-            var semanticModel = syntaxContext.SemanticModel;
```

Comment:
It's unclear to me why so much logic changed in this method.  Can you explain what your new code does that's different from teh old way of checking def assignment?  Thanks!